Source/DafnyStandardLibraries/src/Std/Collections/Tuple.dfy

```diff
@@ -0,0 +1,19 @@
+/*******************************************************************************
+ *  Copyright by the contributors to the Dafny Project
+ *  SPDX-License-Identifier: MIT
+ *******************************************************************************/
+
+/**
+ This module defines useful properties and functions relating to the built-in tuple type.
+ */
+module Std.Collections.Tuple {
+  /** Extracts the first component of a pair */
+  function T2_0<L, R>(): ((L, R)) -> L {
```

Comment:
Why not something more readable like `First`/`Second`?

Reply:
Because we can't make this function polymorphic in the arity of the tuple, and a tuple can contain more elements than a pair. We wouldn't be able to reuse `First` and `Second` for tuples of arity 3.
Since this is the standard library, I'd prefer to have a notation that will work no matter the arity of the tuple. By including the arity of the tuple in the function's name, it's very easy to generalize.

Reply:
I do occasionally wish we had overloading :) 

I guess this is here more so that you can use `T2_0` as a function value, rather than using `T2_0(p)` instead of the simpler `p.0`? I can accept that then. There are also functions like `Seq.First` which read better than the natural syntax, which is what I thought you were after at first.